### ISSUE TITLE
Attestation report via TSM

### ISSUE THREAD

**Issue by IT302** [coconut-svsm/svsm#418]
I am unable to generate an attestation report using TSM. I have tried:
```
x:~# report=/sys/kernel/config/tsm/report/report0
x:~# mkdir $report
x:~# dd if=/dev/urandom bs=64 count=1 > $report/inblob
x:~# hexdump -C $report/outblob
hexdump: /sys/kernel/config/tsm/report/report0/outblob: Invalid argument
```

```
x:~# ls -la $report/
total 0
drwxr-xr-x 2 root root    0 Jul 22 10:07 .
drwxr-xr-x 4 root root    0 Jul 22 10:07 ..
-r--r--r-- 1 root root    0 Jul 22 10:20 auxblob
-r--r--r-- 1 root root 4096 Jul 22 10:20 generation
--w------- 1 root root    0 Jul 18 16:17 inblob
-r--r--r-- 1 root root    0 Jul 22 10:20 manifestblob
-r--r--r-- 1 root root    0 Jul 22 10:20 outblob
--w------- 1 root root 4096 Jul 18 16:17 privlevel
-r--r--r-- 1 root root 4096 Jul 22 10:20 privlevel_floor
-r--r--r-- 1 root root 4096 Jul 22 10:20 provider
--w------- 1 root root 4096 Jul 22 10:05 service_guid
--w------- 1 root root 4096 Jul 18 14:11 service_version
--w------- 1 root root 4096 Jul 22 10:04 svsm
```
I have also tried using the new interface for SVSM, but no luck either. 
Could someone please help me figure out what I am missing?

**Comment by tlendacky** [comment]
> x:~# report=/sys/kernel/config/tsm/report/report0
> x:~# mkdir $report
> x:~# dd if=/dev/urandom bs=64 count=1 > $report/inblob
> x:~# hexdump -C $report/outblob
> hexdump: /sys/kernel/config/tsm/report/report0/outblob: Invalid argument

Since you're running under an SVSM, you can't request a privlevel less than the VMPL at which you are running. Try adding:
echo 2 > /sys/kernel/config/tsm/report/privlevel

As for the SVSM attestation, I don't believe coconut-svsm has implemented the attestation protocol, yet.

**Comment by IT302** [comment]
I can confirm that adding "echo  2 > $report/privlevel"  made it work.  Thank you @tlendacky for your help.

**Comment by tlendacky** [comment]
The SVSM guest support that is being merged into Linux 6.11 will provide a sysfs file that will contain the current VMPL at which Linux is running. That can be used to set the privlevel appropriately:

cat /sys/devices/system/cpu/sev > $report/privlevel
